src/FileFetcher.php

```diff
@@ -27,22 +28,37 @@ public function __construct($filePath, $temporaryDirectory = "/tmp")
         $state['destination'] = $file->isFile() ? $filePath : $this->getTemporaryFilePath($filePath);
 
         if (!$file->isFile() && $this->serverIsNotCompatible($filePath)) {
-            throw new \Exception("The server hosting the file does not support ranged requests.");
+            $this->compatibleServer = false;
+            $state['total_bytes'] = PHP_INT_MAX;
+            $this->deleteFile($state['destination']);
+        } else {
+            $state['total_bytes'] = $file->isFile() ? $file->getSize() : $this->getRemoteFileSize($filePath);
         }
 
-        $state['total_bytes'] = $file->isFile() ? $file->getSize() : $this->getRemoteFileSize($filePath);
-
         if (file_exists($state['destination'])) {
             $state['total_bytes_copied'] = filesize($state['destination']);
         }
 
         $this->setState($state);
     }
 
+    public function setTimeLimit(int $seconds)
```

Comment:
Document! Also this seems a little strange... perhaps the timelimit should be set to PHP_INT_MAX by default, and this function should just fail gracefully if compatibleServer = false?